src/Application.cpp

```diff
@@ -116,8 +116,13 @@ int Application::run(QApplication &qtApp)
 
     this->windows->getMainWindow().show();
 
+#if defined(Q_OS_WIN) || defined(Q_OS_LINUX) || defined(Q_OS_MACOS)
```

Comment:
I would rather have this check in a function inside the Updates.hpp/cpp so all the update logic is grouped in one file

Reply:
The reason why I did it this way is, that otherwise the signal will be connected, resulting in unnecessary calls to `checkForUpdates`. For that reason, having that check at those two places isn't as bad as you claim, but if you think that it's better to have an immediate return in checkForUpdates, I'll respect that.